### PR TITLE
Add filter version of fluent-plugin-filter

### PR DESCRIPTION
**Pull request by cosmo0920**
Hi!

Thanks for providing great fluentd plugin!!
I think this plugin is one of the suitable case to provide filter plugin which is introduced since Fluentd v0.12.
I added filter version of `fluent-plugin-filter`.
In more detail, please refer to the official Fluentd.org document: http://docs.fluentd.org/articles/filter-plugin-overview.

And I've found that original output type fluent-plugin-filter testing code contains potential anti-pettern unit  testing.

More concretely, when using one by one assertions in tests, if one of the assertion is failed, after assertion is *nothing to be executed*. This makes hard to find out potential unit testing problem.
In contrast, data driven tests will be translated to individual test case per data by unit-test framework.
When using this unit testing way, once one of the test case will be failed, other test cases will be certainly executed.
In conclusion, data driven testing is more suitable than one by one assertions testing for this plugin testing code.

Enjoy fluentd plugin development with Ruby 2.2 and new filter plugin functionality.

Cheers!

### CODE REVIEW

**Comment by muddydixon** [comment]
I'm sorry for late to response.

Now review and check it!
 
Thanks a lot.

**Comment by muddydixon** [comment]
release v0.0.4 https://rubygems.org/gems/fluent-plugin-filter

**Comment by cosmo0920** [comment]
Awesome! Thanks.
